account-settings/set_user_preferences.py

```diff
@@ -42,30 +57,22 @@ def login(login_url):
 def get_users(i):
     print("Actual page")
     print(i)
-    login(login_url)
     post_url = url+"/api/v4/users"
     payload = {'page': i, 'per_page': '200'}
     response = requests.get(post_url, headers=hed, params=payload)
     users = response.json()
     for user in users:
-        for k,v in user.items():
-            if k == "id":
-                user_id = v
-                set_prefs(user_id)
-                get_prefs(user_id)
+        print("setting preferences for user: "+user['username'])
+        user_id = user['id']
+        set_prefs(user_id)
+        # get_prefs(user_id) - optional, just for testing
 
 def get_prefs(id):
-    login(login_url)
-    print("ID: "+id)
     post_url = url+"/api/v4/users/"+id+"/preferences"
     response = requests.get(post_url, headers=hed)
     info = response.json()
-    print(info)
-    for i in info:
-        print(i)
 
 def set_prefs(id):
```

Comment:
I note that we clobber any existing fields in `sidebar_settings`, but I think this is probably fine, since this is specific to this preference and isn't likely to change in the experimental version anyway.